### ISSUE TITLE
Cannot start application on Raspbian Stretch

### ISSUE THREAD

**Issue by T94T** [joshmarinacci/aminogfx-gl#13]
My application does not work anymore on a newly installed Raspbian Stretch. I get this error:
module.js:664

> 
>   return process.dlopen(module, path._makeLong(filename));
>                  ^
> 
> Error: /home/pi/movieplayer/node_modules/aminogfx-gl/build/binding/aminonative.node: file too short
>     at Object.Module._extensions..node (module.js:664:18)
>     at Module.load (module.js:554:32)
>     at tryModuleLoad (module.js:497:12)
>     at Function.Module._load (module.js:489:3)
>     at Module.require (module.js:579:17)
>     at require (internal/module.js:11:18)
>     at Object.<anonymous> (/home/pi/movieplayer/node_modules/aminogfx-gl/main.js:14:16)
>     at Module._compile (module.js:635:30)
>     at Object.Module._extensions..js (module.js:646:10)
>     at Module.load (module.js:554:32)

BTW: I had to install libgles2-mesa-dev manually. Else npm install throws an error.

**Comment by joshmarinacci** [comment]
hm.. is this still on ARM? I wonder if any lib dirs changed.

-j

**Comment by T94T** [comment]
Yes, I'm using it on my Raspberry Pi Zero W.

**Comment by cbratschi** [comment]
Which Node version are you using? I am using Stretch with Node 7.10.1 for instance.

Does a rebuild work? Looks like a library mismatch.

**Comment by gepatto** [comment]
In Raspberry Stretch the GLES libs have been renamed, to avoid name collisions with non RPI-specific linux gles libraries. They renamed *libEGL.so* and *libGLESv2.so* to *libbrcmEGL.so* and *libbrcmGLESv2.so* so you will will need to link to the new libs on stretch.

**Comment by T94T** [comment]
Thank you! I replaced the EGL and GLES attributes in binding.gyp with -lbrcmEGL and -lbrcmGLESv2. After that
```
rm -rf node_modules/aminogfx-gl/build
npm rebuild
```

Now it works!

(I'm using nodejs v8.9.3 and npm 5.5.1)

**Comment by cbratschi** [comment]
Thanks for this details. I upgraded our system to Stretch and it seems the original files were kept together with the renamed ones. Therefore builds were still running.

I will add the Stretch settings to the build file. However, I could not find a way to support Jessie and Stretch in parallel using Gyp. So it will be Jessie by default for now. Hopefully there is a better way to support optional libraries or filter by OS subversion.

Closing the thread. Feel free to comment if you know a way to support Jessie and Stretch in a single Gyp file.
